Add GitHub App Installation Guidelines document (#412)
hiero-ledger/governance: guidelines/gh-app-installs.md

# GitHub App Installation Guidelines

Installing GitHub Apps allows automation and integration, but they can potentially be a security risk since each app can access parts of your repositories and organization data.
Follow these steps to ensure you install apps securely and intentionally.

## Research Your App's Legitimacy

- Make sure the app belongs to a trusted publisher.

> - Read reviews in GitHub Marketplace if available or do a short research.

Reviewer: ```suggestion
- Read reviews in GitHub Marketplace if available and review other sources of feedback.
```

> - Inspect the permissions required as well as potential costs.

Author: I kinda wanna include the word costs too.. since some apps are paid and we need to spot that.

> - Review if there is any other similar app already installed in your organization and available for your repos.

Author: This one is tricky, since sometimes we will install apps in the org with permissions to 1 repo (for example). So we need to make sure people check if already existing apps in the org are available that can just be extended to the repo in the request. While there could be a similar app already installed in their repo, it could also be the case that similar apps are installed in other repos but not in theirs.

## Validate Functionality

- Ensure that the app works as intended (in your test fork).
- Avoid requesting apps that could potentially cause conflict with already existing apps.
- After installation (and if approved) make sure to monitor and report any strange behavior.

## Scope the App's needs

- Scope and limit the repos in your organization that need the app.

> - Ensure that the permissions required for the app to function align with your organization's guidelines.

Reviewer: ```suggestion
- Ensure the permissions required for the app to function align with your organization's guidelines.
```
